contracts/contracts/vault/VaultCore.sol

```diff
@@ -97,6 +107,46 @@ contract VaultCore is VaultStorage {
         }
     }
 
+    /**
+     * @dev Mint OUSD for OUSD Meta Strategy
+     * @param _amount Amount of the asset being deposited
+     *
+     * Notice: can't use nonReentrant modifier since BaseCurveStrategy's deposit
+     * already has that modifier present
+     */
+    function mintForStrategy(uint256 _amount)
+        external
+        whenNotCapitalPaused
+        onlyOusdMetaStrategy
+    {
+        require(_amount > 0, "Amount must be greater than 0");
+
+        emit Mint(msg.sender, _amount);
+
+        // Rebase must happen before any transfers occur.
+        // TODO: double check the relevance of this
+        if (_amount >= rebaseThreshold && !rebasePaused) {
+            _rebase();
+        }
+
+        netOusdMintedForStrategy += int256(_amount);
+
+        require(
+            abs(netOusdMintedForStrategy) < netOusdMintForStrategyThreshold,
+            string(
+                bytes.concat(
+                    bytes("Attempting to mint too much OUSD. Threshold: "),
```

Comment:
We should probably save contract size and not build these errors with numbers in them.

Reply:
Cool, removed. Influence of UniswapV2Pair Tech makes it even more compelling to compact the code.